gwi-project/src/components/Dog/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { SetStateAction, useEffect, useState } from "react"
+import { DogImage } from "../DogImage"
+import { useLocation, useNavigate } from "react-router-dom"
+import { css } from "@emotion/css"
+import { Pagination } from "../Pagination"
+import { capitalizeFirstLetter } from "../../Utils/stringUtils"
+import { globalBreadcrumbStyles } from "../DogBreeds"
+import { getBreedImages } from "../../api/record"
+
+export function Dog() {
+  const [breedImages, setBreedImages] = useState<string[]>([])
+  const location = useLocation()
+  const breed = location.pathname.split("/")[2]
+  const pageQuery = location.search.split("=")[1]
+  const paginate = (pageNumber: SetStateAction<number>) => setOffset(pageNumber)
+  const page =
+    pageQuery && !isNaN(parseInt(pageQuery)) ? parseInt(pageQuery) : 1
+  const [offset, setOffset] = useState<number>(page)
+  const dogsPerPage = 6
+  const navigate = useNavigate()
+  useEffect(() => {
+    navigate(`?currentPage=${offset}`)
+  }, [offset])
+
+  //I do not like client pagination, but I do not see anywhere in the documentation
+  // mentioning offset and limit...
+  useEffect(() => {
+    getBreedImages(breed)
+      .then((response) => {
+        setBreedImages(response.message)
+      })
+      .catch((error) => {
+        console.error(error)
+      })
+  }, [breed])
+
+  const currentPageDogs = breedImages?.slice(
+    (offset - 1) * dogsPerPage,
+    offset * dogsPerPage
+  )
+  return (
+    <div>
+      <span
+        data-testid="goBackContainer"
+        className={globalBreadcrumbStyles.goBackContainer}
+      >
+        <span
+          data-testid="mainPage"
+          className={globalBreadcrumbStyles.goBack}
+          onClick={() => {
```

Comment:
Wouldn't it be better to have the handler functions on the top of the component?

Reply:
Yes, I agree.